modules/3d/src/pointcloud/io_ply.cpp

```diff
@@ -142,42 +310,184 @@ T readNext(std::ifstream &file, DataFormat format)
     return val;
 }
 
-void PlyDecoder::parseBody(std::ifstream &file, std::vector<Point3f> &points, std::vector<Point3f> &normals, std::vector<Point3_<uchar>> &rgb)
+template <>
+uchar readNext<uchar>(std::ifstream &file, DataFormat format)
+{
+    int val {0};
+    if (format == DataFormat::ASCII)
+    {
+        file >> val;
+    }
+    else
+    {
+        file.read((char *)&val, sizeof(uchar));
+    }
+    return (uchar)(val & 0xff);
```

Comment:
The solution looks suspicious.
```
file.read((char *)&val, sizeof(uchar));
```
Depends on endianess. On big-endian you'll get something 0xFF000... and the result will be always zero.

Reply:
Sure, gonna fix it